subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/DefaultFileWatcherRegistry.java

```diff
@@ -86,15 +87,20 @@ public void handleChangeEvent(FileWatchEvent.ChangeType type, String absolutePat
 
                             @Override
                             public void handleUnknownEvent(String absolutePath) {
+                                LOGGER.error("Received unknown event for {}", absolutePath);
                                 fileWatchingStatistics.updateAndGet(MutableFileWatchingStatistics::unknownEventEncountered);
-                                handler.handleLostState();
+                                handler.stopWatchingAfterError();
                             }
 
                             @Override
                             public void handleOverflow(FileWatchEvent.OverflowType type, @Nullable String absolutePath) {
                                 if (absolutePath == null) {
-                                    handler.handleLostState();
+                                    LOGGER.info("Overflow detected, invalidating all watched hierarchies");
```

Comment:
Do we do some logging on the native side as well? Do we maybe need to change the log-level there?

Reply:
We log at `INFO` on the native side:

https://github.com/gradle/native-platform/blob/24b02d4d6022d8ed6a6986c1ae9e1e77c4951d6f/file-events/src/file-events/cpp/generic_fsnotifier.cpp#L83-L89

But the messages cover different purposes, so I think it's okay to log both.